crates/router/src/core/payments/helpers.rs

```diff
@@ -280,9 +280,13 @@ pub fn validate_request_amount_and_amount_to_capture(
 
 pub fn validate_mandate(
     req: impl Into<api::MandateValidationFields>,
-) -> RouterResult<Option<api::MandateTxnType>> {
+) -> CustomResult<Option<api::MandateTxnType>, errors::ApiErrorResponse> {
     let req: api::MandateValidationFields = req.into();
-    match req.is_mandate() {
+    match req
+        .is_mandate()
```

Comment:
Change the function name for `is_mandate()`. We are doing change_context and doing ?. And the function name doesn't express what it does.